_posts/2025-02-05-#31363.md

```diff
@@ -10,17 +10,263 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+### General Cluster Mempool
+
+Skip this section if you are just interested in reviewing the code and don't need to hear the pitch for why.
+
+The primary goal of [Cluster Mempool](https://github.com/bitcoin/bitcoin/issues/30289) is to enable the mempool to more
+intelligently assess which transactions would be most incentive compatible to keep. In simple terms "which transactions
+would earn a miner the most in fees?" The complexity of incentive compatibility assessment comes from dependencies
+between transactions, i.e. spending relationships.
+
+If the motivations for this project are unclear to you, try to spend up to 60min on these reading materials:
+
+- "But aren't fees a trivially small portion of block rewards? Why do we care so much?"
+  - One of the biggest problems we want to solve is the asymmetry between how we decide what to evict and what goes into blocks: it's possible
+    that [the first transaction we would evict is also the first transaction we would mine](https://github.com/bitcoin/bitcoin/issues/27677#eviction-mining-not-opposite).
+  - An original motivation for trying to come up with a metric for "incentive compatibility" was within the context of
+    [RBF pinning](https://gist.github.com/glozow/25d9662c52453bd08b4b4b1d3783b9ff#mining-score-of-a-mempool-transaction); our
+current feerate-based approach causes us to reject transactions that would be more incentive compatible, creating
+censorship problems.
+
+- "Why is cluster-tracking the solution for incentive compatibility?"
+  - [This Delving post](https://delvingbitcoin.org/t/mempool-incentive-compatibility/553) provides a great explanation of why the question of incentive
+    compatibility is complicated and why tracking clusters is necessary for a complete solution.
+  - One of the benefits of Topologically Restricted Until Confirmation is to have an effective cluster limit within
+the existing framework of ancestor/descendant limits (limit of cluster count 2 is the equivalent of limiting ancestor
+and descendant counts to 2). However, a limit of 2 is quite prohibitive.
+  - We talked about clusters in previous review club meetings about [auto-bumping unconfirmed UTXOs in coin selection](/26152)
+    (which added `CTxMmemPool::GatherClusters`) and [v3 transactions](/25038).
+
+Mempool transactions and their dependency relationships can naturally be represented as a directed graph where nodes are
+transactions and edges represent UTXO spending relationships. A transaction's _cluster_ is the part of the graph that it
+is connected to.
```

Comment:
I think there's some potential for naming ambiguity here, that might be worth touching on? If my understanding is correct:
A cluster is the set of all nodes that are indirectly connected to each other through UTXO spending relationships. I think that generally also falls under the definition of a graph? So "A transaction's _cluster_ is the part of the graph that it is connected to." can perhaps be a bit confusing? I think a cluster is a graph, and `TxGraph` represents all the graphs/clusters that exist in the mempool?

I don't think this OP has to define the terms, but I think highlighting this might help avoid confusion?